build-system-tests/scripts/mega-app-copy-files.sh

```diff
@@ -124,8 +124,8 @@ if [[ "$FRAMEWORK" == 'react' && "$BUILD_TOOL" == 'vite' ]]; then
     # https://ui.docs.amplify.aws/react/getting-started/troubleshooting#vite
     echo "cp templates/components/react/vite/index.html mega-apps/${MEGA_APP_NAME}/index.html"
     cp templates/components/react/vite/index.html mega-apps/${MEGA_APP_NAME}/index.html
-    echo "cp templates/components/react/vite/template-tsconfig-vite-${BUILD_TOOL_VERSION}.json mega-apps/${MEGA_APP_NAME}/tsconfig.json"
-    cp templates/components/react/vite/template-tsconfig-vite-${BUILD_TOOL_VERSION}.json mega-apps/${MEGA_APP_NAME}/tsconfig.json
+    echo "cp templates/components/react/vite/template-tsconfig-vite-${BUILD_TOOL_VERSION}.json mega-apps/${MEGA_APP_NAME}/tsconfig.app.json"
+    cp templates/components/react/vite/template-tsconfig-vite-${BUILD_TOOL_VERSION}.json mega-apps/${MEGA_APP_NAME}/tsconfig.app.json
```

Comment:
Based on this commit: https://github.com/vitejs/vite/pull/15913/files#diff-4aee86d836cbe0aac0d45ef61a8ca540455ef2a7ee667abb3dc1fdac0956b6a5, it looks like we want to overwrite the `tsconfig.app.json` now.

`tsconfig.json` should only references to `tsconfig.app.json` and `tsconfig.node.json`